src/N3Lexer.js

```diff
@@ -471,7 +478,14 @@ export default class N3Lexer {
           }
           // Otherwise, tokenize as far as possible
           else {
-            this._input += data;
+            const dataAsString = String(data);
+            if (this._dataPrependTimes === 0) {
```

Comment:
Note that we could have the pathological case that the first input chunk is an empty string, and then the BOM mark comes. So `this._bomMarkChecked` (see below) should only be set once _some_ first character has been read.

Reply:
Wow, hadn't thought about that. Thanks